### DISCUSSION TITLE
Docker 1.27  - Error - Downgrade 1.26

### DISCUSSION THREAD

**Discussion by ghost** [freelawproject/courtlistener#1556] [Q&A]
The docker-compose.yml files only work up to Docker-Composer 1.26 (not 1.27).
1. Install Docker-2-3-0-4-DesktopInstaller.exe (no higher version - else you get a cpus string error) 
2. There you have first to run :  docker network create cl_net_overlay
3. then run in C:\xxxxxxx\courtlistener\docker\courtlistenerdocker\docker-compose.yml -compose up
4. how to resolve the silenium error?
5. 
    http://127.0.0.1:8000 - Your dev homepage
    http://127.0.0.1:8000/admin - The Django admin page (try the super user)
    http://127.0.0.1:8983/solr - Solr admin page
    127.0.0.1:5900 - A VNC server to the selenium machine (it doesn't serve http though)
None of these is accessible.

Axel Arnold Bangert - Herzogenrath 30.01.2021

![DockerCourtlistener-SileniumError](https://user-images.githubusercontent.com/11486866/106355746-41c97580-62fa-11eb-9822-c05c0f874226.png)

**Comment by mlissner** [comment]
That screenshot looks like an error with windows. You might check the
documentation for that path and see if it can be changed. It'll come up if
you Google it for selenium.

**Reply by ghost** [reply]
Thank you for your answer. The windows dev/shm file exists problem is discussed many times without a real solution. 
Tomorrow I'll try to change the folder.

**Reply by ghost** [reply]
Hi Mr. Lissner,
il y a seulement une passage de code où le dossier est définé et utilisé. J'ai changer le lieu de dossier ...  - ${CL_SHM_DIR:-/dev/shm1}:/dev/shm1 ...  maintenant naturellment il n y plus l'erreur de protection securité, Mais il faut voir s'il est trouvé par les autres container. Selenium est courent a http://127.0.0.1:4444/. 

Mais plusieurs cl-solr java composants ne pouvait pas etre chargés. Je faisait ... docker pull solr ... mais sans effect (img01).
Il y a des erreurs comme REDIS_HOST,  REDIS_PORT etc. c'est pourquoi j'ai inserté le code de 05-private.example dans l'initialization de 10-public.py et Django était trouvable a 8000 en debug-mode (img02) et mode-production (img03).

The default admin user is :  recap 
But the password for this defalult administration user ... recap ... is unknown and undocumented. I could not find it anywhere. So - for access (login) of the admin page http://127.0.0.1/admin you can add a new superuser e.g. like that:

docker exec cl-django /opt/courtlistener/manage.py shell -c "from django.contrib.auth import get_user_model; User = get_user_model(); User.objects.create_superuser('admin2', 'admin2@admin2.de', 'admin2')"

So after that you can logon as admin2 (img04) into the admin panel of courtlistener. But the Docker-Images of Courtlistener in Windows (though linux containers) are incomplete. That is why Windows is not supported for development by Courtlistener and that is OK :) .
Axel

(img01)
![DockerCourtlistener-SileniumErrorFolderTest03](https://user-images.githubusercontent.com/11486866/106381418-d1355e00-63b8-11eb-8a59-6399aecd9c59.png)

(img02)
![DockerCourtlistener-SileniumErrorFolderTest04](https://user-images.githubusercontent.com/11486866/106383426-2a57be80-63c6-11eb-867c-b5abb5d4c963.png)

(img03)
![DockerCourtlistener-SileniumErrorFolder-Production-Mode-Test05](https://user-images.githubusercontent.com/11486866/106385378-52e4b600-63d0-11eb-823d-f20e51e936b2.png)

(img04)
![DockerCourtlistener-SileniumErrorFolder-Admin-Test06](https://user-images.githubusercontent.com/11486866/106625197-bf011e80-6576-11eb-807b-e8a931e3ceb4.png)

**Reply by mlissner** [reply]
We would probably be wise to stick with English even though I've got my phone on French, but if I'm reading this right, it sounds like things are working for you? What are you working on building? Nobody has ever gotten this working in Windows before. It's fairly unsupported, but maybe it just works!

**Reply by ghost** [reply]
Dear Mr. Lissner,
as you proposed I may like to try to open a PR https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request , but I am not sure, whether I have enough detailed information to make that effective. I never placed a PR :)
Best thanks
Axel
